src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/sql.js

```diff
@@ -0,0 +1,395 @@
+    }
+
+    onDataReceived(data) {
+        while (this.dom.firstChild) {
+            this.dom.removeChild(this.dom.lastChild)
+        }
+
+        let parsedData = data
+        if (typeof data === 'string') {
+            parsedData = JSON.parse(data)
+        }
+
+        let visHtml = visualizationStyle
+        if (parsedData.error !== undefined) {
+            visHtml += parsedData.error
+        } else {
+            const params = parsedData.interpolations.map(param =>
+                renderInterpolationParameter(this.theme, param)
+            )
+
+            let language = 'sql'
+            if (parsedData.dialect == 'postgresql') {
+                language = 'postgresql'
+            }
+
+            const formatted = sqlFormatter.format(parsedData.code, {
+                params: params,
+                language: language,
+            })
+
+            const codeRepresentation = '<pre class="sql">' + formatted + '</pre>'
+            visHtml += codeRepresentation
+        }
+
+        const containers = this.createContainers()
+        const parentContainer = containers[0]
+        const scrollable = containers[1]
+        scrollable.innerHTML = visHtml
+        this.dom.appendChild(parentContainer)
+
+        const tooltip = new Tooltip(parentContainer)
+        const baseMismatches = this.dom.getElementsByClassName('mismatch')
+        const extendedMismatchAreas = this.dom.getElementsByClassName('mismatch-mouse-area')
+        setupMouseInteractionForMismatches(tooltip, baseMismatches)
+        setupMouseInteractionForMismatches(tooltip, extendedMismatchAreas)
+    }
+
+    /**
+     * Creates containers for the visualization.
+     */
+    createContainers() {
+        const parentContainer = document.createElement('div')
+        parentContainer.setAttributeNS(null, 'style', 'position: relative;')
+        const width = this.dom.getAttributeNS(null, 'width')
+        const height = this.dom.getAttributeNS(null, 'height')
+        const scrollable = document.createElement('div')
+        scrollable.setAttributeNS(null, 'id', 'vis-sql-view')
+        scrollable.setAttributeNS(null, 'class', 'scrollable')
+        scrollable.setAttributeNS(null, 'viewBox', '0 0 ' + width + ' ' + height)
+        scrollable.setAttributeNS(null, 'width', '100%')
+        scrollable.setAttributeNS(null, 'height', '100%')
+        const viewStyle = `width: ${width - 5}px;
+             height: ${height - 5}px;
+             overflow: scroll;
+             padding:2.5px;`
+        scrollable.setAttributeNS(null, 'style', viewStyle)
+        parentContainer.appendChild(scrollable)
+        return [parentContainer, scrollable]
+    }
+
+    setSize(size) {
+        this.dom.setAttributeNS(null, 'width', size[0])
+        this.dom.setAttributeNS(null, 'height', size[1])
+    }
+}
+
+/**
+ * Splits a qualified type name into a module prefix and the typename itself.
+ */
+function splitQualifiedTypeName(name) {
```

Comment:
I would name it `splitModuleNae` instead and return object containing field `moduleName` instead of "prefix". This function name sounds like it splits qualified type name, and it does not do it.

Reply:
I called it based on what it takes instead of what it returns.

I was thinking about `moduleName` result, but please note that the `prefix` also includes the trailing dot (so `prefix = "Foo.Bar."` and for `moduleName` that dot would not really be part of a module name). This design is simpler because I need to split just to be able to apply different styles to these componenents and so I want to have this dot as part of the prefix. If I didn't return the prefix with the dot, I would need to separately handle the case of an unqualified name which has no prefix (to avoid adding an unnecessary dot there), so I think it would slightly complicate the design.